BUILD.md

```diff
@@ -0,0 +1,116 @@
+# Development Guide
+
+This document provides instructions on how to build and run Kanister locally.
+
+## Architecture
+
+Kanister is a data management framework written in Go. It allows users to
+express data protection workflows using blueprints and actionsets. These
+resources are defined as Kubernetes
+[Custom Resource Definitions](https://docs.kanister.io/architecture.html#custom-resources)
+, following the operator pattern.
+
+![kanister workflow](./graphic/kanister_workflow.png)
+
+## Repository Layout
+
+* `build` - A collection of shell scripts used by the Makefile targets to build,
+test and package Kanister
+* `cmd` - Go `main` packages containing the source of the `controller`,
+`kanctl` and `kando` executables
+* `docker` - A collection of Dockerfiles for build and demos
+* `docs` - Source of the documentation at docs.kanister.io
+* `examples` - A collection of example blueprints to show how Kanister works
+with different data services
+* `graphic` - Image files used in documentation
+* `helm` - Helm chart for the Kanister operator
+* `pkg` - Go library packages used by Kanister
+
+## Development
+
+The [Makefile](Makefile) provides a set of targets to help simplify the build
+tasks. To ensure cross-platform consistency, many of these targets use Docker
+to spawn build containers based on the `ghcr.io/kanisterio/build` public image.
+
+Use the `check` target to ensure your development environment has the necessary
+development tools:
+
+```sh
+make check
+```
+
+The following targets can be used to lint, test and build the Kanister
+controller:
```

Comment:
should we also add steps to build  `kanctl` and `kando` binaries. If not separately there should be a target that just builds all the binaries.

Reply:
Can be done in a follow up PR 👍🏼

Reply:
I updated `build/build.sh` in https://github.com/kanisterio/kanister/pull/1355 to build `kanctl` and `kando`.